### ISSUE TITLE
Cannot Hive Partition by columns containing `/` characters

### ISSUE THREAD

**Issue by archiewood** [duckdb/duckdb#10575]
### What happens?

If you attempt to hive partition a dataset by a column that has values containing the `/` character, it will fail

```
Error: IO Error: Failed to create directory "bands/band=AC/DC"!
```

It is possible this is intended behaviour, in which case a solution would be to document which characters are not allowed.

### To Reproduce

```sql
create table bands as select 'Metallica' as band union all select 'AC/DC' as band;
copy bands to 'bands' (format parquet, partition_by band);
```

### Workaround

Remove all `/` characters from your partition columns

### OS:

MacOS

### DuckDB Version:

0.9.2

### DuckDB Client:

CLI

### Full Name:

Archie Sarre Wood

### Affiliation:

Evidence.dev

### Have you tried this on the latest [nightly build](https://duckdb.org/docs/installation/?version=main)?

I have tested with a nightly build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by archiewood** [comment]
Arrow appear to solve this issue with URL encoding:

https://github.com/apache/arrow/pull/14646/files

**Comment by archiewood** [comment]
This is still not fixed AFAIK

**Comment by archiewood** [comment]
If the solution here is just documenting that / are illegal then i am happy to contribute it

**Comment by Mytherin** [comment]
Thanks for reporting! I've pushed a fix in #12974
